linux/ibus-keyman/src/engine.c

```diff
@@ -319,6 +319,115 @@ ibus_keyman_engine_init(IBusKeymanEngine *keyman) {
   }
 }
 
+static km_kbp_cp* get_base_layout()
+{
+  return u"en-US";
+
+#if 0  // in the future when mnemonic layouts are to be supported
+  const gchar *lang_env = g_getenv("LANG");
+  gchar *lang;
+  if (lang_env != NULL) {
+    g_message("LANG=%s", lang_env);
+    gchar **splitlang = g_strsplit(lang_env, ".", 2);
+    g_message("before . is %s", splitlang[0]);
+    if (g_strrstr(splitlang[0], "_")) {
+      g_message("splitting %s", splitlang[0]);
+      gchar **taglang = g_strsplit(splitlang[0], "_", 2);
+      g_message("lang of tag is %s", taglang[0]);
+      g_message("country of tag is %s", taglang[1]);
+      lang = g_strjoin("-", taglang[0], taglang[1], NULL);
+      g_strfreev(taglang);
+    }
+    else {
+      lang = g_strdup(splitlang[0]);
+    }
+    g_strfreev(splitlang);
+  }
+  else {
+    lang = strdup("en-US");
+  }
+  g_message("lang is %s", lang);
+  km_kbp_cp *cp = g_utf8_to_utf16(lang, -1, NULL, NULL, NULL);
+  return cp;
+  // g_free(lang);
+#endif
+}
+
+static km_kbp_status
+setup_environment(IBusKeymanEngine *keyman)
+{
+  g_assert(keyman);
+  g_message("%s: setting up environment", __FUNCTION__);
+
+  // Allocate enough options for: 3 environments plus 1 pad struct of 0's
+  km_kbp_option_item *environment_opts = g_new0(km_kbp_option_item, 4);
```

Comment:
Might be better to allocate these on the stack:
```suggestion
  km_kbp_option_item environment_opts[4] = {0};
```